src/lib.rs

```diff
@@ -337,3 +340,27 @@ impl<'lib, T> fmt::Debug for Symbol<'lib, T> {
 
 unsafe impl<'lib, T: Send> Send for Symbol<'lib, T> {}
 unsafe impl<'lib, T: Sync> Sync for Symbol<'lib, T> {}
+
+/// Converts a library name to a filename generally appropriate for use on the system.
+///
+/// The function will prepend prefixes (such as `lib`) and suffixes (such as `.so`) to the library
+/// `name` to construct the filename.
+///
+/// # Examples
+///
+/// It can be used to load global libraries in a platform independent manner:
+///
+/// ```
+/// use libloading::{Library, library_filename};
+/// // Will attempt to load `libLLVM.so` on Linux, `libLLVM.dylib` on macOS and `LLVM.dll` on
+/// // Windows.
+/// let library = Library::new(library_filename("LLVM"));
+/// ```
+pub fn library_filename<S: AsRef<OsStr>>(name: S) -> OsString {
```

Comment:
I removed the conditional logic from this function. Felt way too magic to me. I don’t anticipate people to be passing in random data, its going to be either the full filename or just the library name, at which point they can have the conditional in their code.

Reply:
I just wanted to comment on this one. But that makes totally sense to be honest. And if such a need arises, people can still write it themselves. It's not that complicated and `library_filename` offers a good template that people can take a look at.

The only use-case I could come up with (that I actually encountered) was loading plugins. Some people just have weird naming schemes and prepend/append stuff that is specific to their OS (like `lib`) :)